doc/classes/AnimationMixer.xml

```diff
@@ -15,10 +15,10 @@
 			<param index="0" name="animation" type="Animation" />
 			<param index="1" name="track" type="int" />
 			<param index="2" name="value" type="Variant" />
-			<param index="3" name="object" type="Object" />
-			<param index="4" name="object_idx" type="int" />
+			<param index="3" name="object_id" type="int" />
```

Comment:
This breaks compat technically. Not sure how we want to handle this.

Reply:
If we really want to preserve compat, we can get the object reference and pass it to the virtual method instead of ObjectID. If there is no object anymore, we would pass a nullptr. Not sure if it would compromise any workflow, but don't immediately see downside with this approach (unless user really needs ObjectID of an object that doesn't exist anymore).

Reply:
That makes sense. There might still be a risk that the `Object *` they get will be a `previously freed instance`, but at least in GDScript this can be checked with `is_instance_valid()`.

The other option is to add another method for this that passes an `object_id`, and keep the old method as deprecated (with the behavior we just described).

Reply:
I'd prefer to preserve compatability, but AnimationMixer is almost an abstract class so that's the concern.

Reply:
The following TrackCaches have pointers to more than just `Object`, so they will need to be modified as well for consistency.

- TrackCacheTransform
  - node_3d
  - skeleton
- TrackCacheBlendShape
  - mesh_3d

Then, `_post_process_key_value()` passes `object` and `object_idx` differently for them. For example, `TrackCacheTransform` passes `Skeleton` and `BoneIndex`. And `TrackCacheBlendShape` passes `Mesh3D` and `BlendShapeIdx`.

So the correct change for this part should be `int object_id, int object_sub_idx`.

Reply:
Thanks @TokageItLab, I moved my reply to your comment in the main PR.

Let's keep this thread focused on how to handle the minor compatibility breakage.